Data/SearchEngine/SearchIndex.hs

```diff
@@ -212,6 +213,10 @@ getDocTermIds SearchIndex{docIdMap} docid =
     case docIdMap IntMap.! fromEnum docid of
       DocInfo _ doctermids _ -> doctermids
 
+getKeyDocId :: Ord key => SearchIndex key field feature -> key -> Maybe DocId
+getKeyDocId SearchIndex{docKeyMap} key = Map.lookup key docKeyMap
```

Comment:
This should probable be in the family of "lookup" functions rather than "get" functions, since it's partial. Perhaps `lookupDocKeyDocId`?